### PR TITLE
Remove perl

### PR DESCRIPTION
**Pull request by shlevy**
https://headcounter.org/hydra/jobset/iElectric/nix-without-perl

### CODE REVIEW

**Comment by shlevy** [comment]
@edolstra please merge this one, not the others. Still working on nix-perl package.

**Comment by shlevy** [comment]
Incorporates #1026 #1023 #1018 #997 #981

**Comment by shlevy** [comment]
@edolstra nix-perl added, ready to merge

**Comment by domenkozar** [comment]
Here's a Hydra jobset building this branch: https://headcounter.org/hydra/eval/330058

**Comment by shlevy** [comment]
FYI https://github.com/NixOS/nixpkgs/issues/17697 is relevant to the copy-closure and remote builds tests

**Comment by shlevy** [comment]
@domenkozar Can you use release-16.03 for the nixpkgs checkout? That's what nix master on hydra.nixos.org uses.

**Comment by domenkozar** [comment]
Done

**Comment by edolstra** [comment]
@shlevy One giant PR won't make things easier to review. I prefer the smaller ones.

**Comment by shlevy** [comment]
Can you review on the smaller ones then merge here once it's all OK? There were several non-trivial cleanups.

**Comment by copumpkin** [comment]
Is this basically done? What's left?

**Comment by domenkozar** [comment]
https://github.com/NixOS/nix/pull/1023 https://github.com/NixOS/nix/pull/981

**Comment by shlevy** [comment]
#1027

**Comment by copumpkin** [comment]
Error: infinite loop detected

**Comment by shlevy** [comment]
:grin: whoops, the last 4 commits here aren't covered in the other PRs

**Comment by shlevy** [comment]
@edolstra Rebased on #1023. The `nix-perl` package added here is a good place to add the perl `nix-copy-closure` if you decide we should go that route.

**Comment by shlevy** [comment]
@edolstra rebased!

**Comment by domenkozar** [comment]
@edolstra & @shlevy thank you so much for this hard work! :D

**Comment by shlevy** [comment]
@edolstra ping

**Comment by edolstra** [comment]
Could `nix-perl` be changed into a "regular" CPAN-style Perl package? I.e. with a Makefile.PL (or build.PL) instead of a autoconf/automake build system. That would make dependency handling easier. (E.g. you can just say `requires DBI;`.) It will also allow it to be used in Nixpkgs using `buildPerlPackage`.

**Comment by shlevy** [comment]
I don't really know anything about that system, but I suppose so. The biggest question is whether we can pull in the dependency on the nix libs and do the `nix-instantiate` calls to pull out the nix config from config.nix

**Comment by shlevy** [comment]
@edolstra Any thoughts on this?

**Comment by zimbatm** [comment]
Is that the last blocker for the 1.12 release?

**Comment by shlevy** [comment]
FWIW, I don't think making `nix-perl` a `CPAN` package should be a blocker here. As things currently stand `nix-perl` can be built using standard autoconf tooling.

**Comment by zimbatm** [comment]
Agreed, this is already an improvement. One thing at a time :)

Can you rebase to resolve the merge conflicts?

**Comment by shlevy** [comment]
Once I get approval otherwise, yep! Or I'll just merge it and resolve during the merge.

File: release.nix [hunk 1]
```diff
@@ -102,6 +97,32 @@ let
...
+
+      releaseTools.nixBuild {
+        name = "nix-perl";
+        src = tarball;
+
+        buildInputs =
+          [ (builtins.getAttr system jobs.build) curl bzip2 xz pkgconfig pkgs.perl ]
+          ++ lib.optional stdenv.isLinux libsodium;
+
+	configureFlags = ''
```
Comment (edolstra) [inline release.nix]: Tabs.

Comment (shlevy) [inline release.nix]: Fixed

**Comment by copumpkin** [comment]
\o/ \o/ \o/

**Comment by zimbatm** [comment]
Wooot! Can we get a release now? ^_^

**Comment by tomberek** [comment]
1.11.16 still has a perl dependency, is that correct? I'm trying boostrap from source into arm7l and eventually ran into this:
```
Can't locate Nix/Config.pm in @INC (you may need to install the Nix::Config module) (@INC contains: 
/usr/local/lib/perl5/site_perl/5.20.2/arm-linux-gnueabihf-thread-multi-64int /usr/local/lib/perl5/si
te_perl/5.20.2/arm-linux-gnueabihf-thread-multi-64int /etc/perl /usr/lib/perl/site_perl/5.20.0/ /usr
/lib/perl/site_perl/5.20.0 /usr/lib/perl/vendor_perl/5.20.0/ /usr/lib/perl/vendor_perl/5.20.0 /usr/l
ib/perl/5.20.0/ /usr/lib/perl/5.20.0 /usr/local/lib/site_perl /usr/lib/perl/5.20.0 .) at ./nix-chann
el line 7.
```

**Comment by shlevy** [comment]
Yes, it's only 1.12

**Pull request merged**